File: Python/소프티어/징검다리.py
import sys
input = sys.stdin.readline

n = int(input())
rocks = list(map(int, input().split()))

dp = [1] * n

for i in range(n):
    temp = 0
    for j in range(i):
        if rocks[i] > rocks[j]:
            temp = max(temp, dp[j])
    dp[i] = temp+1

print(max(dp))

File: AlgorithmReview/softeer/징검다리.swift
//
//  징검다리.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2024/02/01.
//

import Foundation

let n = Int(readLine()!)!
let rocks = readLine()!.split(separator: " ")

var dp = [Int](repeating: 0, count: n)
for i in 0..<n {
    var temp = 0
    for j in 0..<i {
        if rocks[i] > rocks[j] {
            temp = max(temp, dp[j])
        }
    }
    dp[i] = temp+1
}

print(dp.max()!)
